ibis/backends/dask/execution/generic.py

```diff
@@ -0,0 +1,869 @@
+    func = getattr(np, type(op).__name__.lower())
+    return call_numpy_ufunc(func, op, data, **kwargs).astype(return_type)
+
+
+def vectorize_object(op, arg, *args, **kwargs):
+    # TODO - this works for now, but I think we can do something much better
+    func = np.vectorize(functools.partial(execute_node, op, **kwargs))
+    out = dd.from_array(func(arg, *args), columns=arg.name)
+    out.index = arg.index
+    return out
+
+
+@execute_node.register(
+    ops.Log, dd.Series, (dd.Series, numbers.Real, decimal.Decimal, type(None))
+)
+def execute_series_log_with_base(op, data, base, **kwargs):
+    if data.dtype == np.dtype(np.object_):
+        return vectorize_object(op, data, base, **kwargs)
+
+    if base is None:
+        return np.log(data)
+    return np.log(data) / np.log(base)
+
+
+@execute_node.register(ops.Ln, dd.Series)
+def execute_series_natural_log(op, data, **kwargs):
+    if data.dtype == np.dtype(np.object_):
+        return data.apply(
+            functools.partial(execute_node, op, **kwargs),
+            meta=(data.name, "object"),
+        )
+    return np.log(data)
+
+
+@execute_node.register(
+    ops.Clip,
+    dd.Series,
+    (dd.Series, type(None)) + numeric_types,
+    (dd.Series, type(None)) + numeric_types,
+)
+def execute_series_clip(op, data, lower, upper, **kwargs):
+    return data.clip(lower=lower, upper=upper)
+
+
+@execute_node.register(ops.Quantile, (dd.Series, SeriesGroupBy), numeric_types)
+def execute_series_quantile(op, data, quantile, aggcontext=None, **kwargs):
+    # TODO - interpolation
+    return data.quantile(q=quantile)
+
+
+@execute_node.register(ops.MultiQuantile, dd.Series, collections.abc.Sequence)
+def execute_series_quantile_sequence(
+    op, data, quantile, aggcontext=None, **kwargs
+):
+    # TODO - interpolation
+    # TODO - to list?
+    return list(data.quantile(q=quantile))
+
+
+@execute_node.register(
+    ops.MultiQuantile, SeriesGroupBy, collections.abc.Sequence
+)
+def execute_series_quantile_groupby(
+    op, data, quantile, aggcontext=None, **kwargs
+):
+    def q(x, quantile, interpolation):
+        result = x.quantile(quantile, interpolation=interpolation).tolist()
+        res = [result for _ in range(len(x))]
+        return res
+
+    result = aggcontext.agg(data, q, quantile, op.interpolation)
+    return result
+
+
+@execute_node.register(
+    ops.Round, dd.Series, (dd.Series, np.integer, type(None), int)
+)
+def execute_round_series(op, data, places, **kwargs):
+    if data.dtype == np.dtype(np.object_):
+        return vectorize_object(op, data, places, **kwargs)
+    result = data.round(places or 0)
+    return result if places else result.astype('int64')
+
+
+@execute_node.register(ops.TableColumn, (dd.DataFrame, DataFrameGroupBy))
+def execute_table_column_df_or_df_groupby(op, data, **kwargs):
+    return data[op.name]
+
+
+@execute_node.register(ops.Aggregation, dd.DataFrame)
+def execute_aggregation_dataframe(
```

Comment:
This xfails?

Reply:
It's a bit hard for me to review from here because not clear to me what works and what should be ignored because lots of these are aggregations..